cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,6 +40,7 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+	SelfHosted        bool
```

Comment:
We should start documenting these types.  Can you add a description here and mark this as alpha?